ludwig/datasets/mixins/download.py

```diff
@@ -46,6 +46,41 @@ def download_raw_dataset(self):
     def download_urls(self):
         return self.config["download_urls"]
 
+
+class GZipDownloadMixin:
+    """Downloads the gzip archive file containing the training data and extracts the contents."""
+
+    config: dict
+    raw_dataset_path: str
+    raw_temp_path: str
+
+    def download_raw_dataset(self):
+        """
+        Download the raw dataset and extract the contents of the zip file and
+        store that in the cache location.
+        """
+        """
+                Download the raw dataset and contents of the gzip file
+                onto the _raw directory.
+                """
+        os.makedirs(self.raw_temp_path, exist_ok=True)
+        for file_download_url in self.download_urls:
+            filename = file_download_url.split('/')[-1]
+            urllib.request.urlretrieve(file_download_url, os.path.join(self.raw_temp_path, filename))
+            size_of_original_file = len(filename)
+            gzip_content_file = filename[:size_of_original_file - 3]
+            with gzip.open(os.path.join(self.raw_temp_path, filename)) as gzfile:
+                file_content = gzfile.read()
```

Comment:
We should be able to avoid reading the entire uncompressed file into memory using the approach shown here: https://stackoverflow.com/a/44712152

So something like:

```
with gzip.open(os.path.join(self.raw_temp_path, filename)) as gzfile:
    with open(os.path.join(self.raw_temp_path, gzip_content_file), 'wb') as output:
        shutil.copyfileobj(gzfile, output)
```

Reply:
Changed, see latest MR